src/config/vars.rs

```diff
@@ -4,7 +4,29 @@ use std::collections::HashMap;
 /// (result, warnings)
 pub fn interpolate(input: &str, vars: &HashMap<String, String>) -> (String, Vec<String>) {
     let mut warnings = Vec::new();
-    let re = Regex::new(r"\$\$|\$(\w+)|\$\{(\w+)(?::-([^}]+)?)?\}").unwrap();
+    // Environment variable names can have any characters from the Portable Character Set other
+    // than NUL
```

Comment:
I was also reading The Open Group specification about environment variable names yesterday (after the issue this PR closes), and it says:
> These strings have the form name=value; names shall not contain the character '='.

Therefore `=` is also not a valid character together with NUL.

Reply:
Good call! I excluded it in `6574ba31d`